rrrspec-client/lib/rrrspec/client/rspec_runner_factory.rb

```diff
@@ -0,0 +1,24 @@
+require 'rrrspec/client/rspec_runner'
+
+module RRRSpec
+  module Client
+    class RSpecRunnerFactory
+      attr_reader :next_factory
+
+      def initialize(successor)
+        @next_factory = successor
+      end
+
+      def create(file_extension)
+        unless Thread.current[:rspec3_runner]
+          Thread.current[:rspec3_runner] = RSpecRunner.new
```

Comment:
`Thread.fork` is for threads while `Kernel.fork` or `Process.fork` is for processes. They're totally different constructs, and I'm not sure what your intention is for this question...